library/modules/Textures.cpp

```diff
@@ -41,205 +48,236 @@ static long g_window_borders_texpos_start = -1;
 //
 // It uses the same pixel format (RGBA, R at lowest address) regardless of
 // hardware.
-SDL_Surface * canonicalize_format(SDL_Surface *src) {
-  SDL_PixelFormat fmt;
-  fmt.palette = NULL;
-  fmt.BitsPerPixel = 32;
-  fmt.BytesPerPixel = 4;
-  fmt.Rloss = fmt.Gloss = fmt.Bloss = fmt.Aloss = 0;
+SDL_Surface* canonicalize_format(SDL_Surface* src) {
+    SDL_PixelFormat fmt;
+    fmt.palette = NULL;
+    fmt.BitsPerPixel = 32;
+    fmt.BytesPerPixel = 4;
+    fmt.Rloss = fmt.Gloss = fmt.Bloss = fmt.Aloss = 0;
 #if SDL_BYTEORDER == SDL_BIG_ENDIAN
-  fmt.Rshift = 24; fmt.Gshift = 16; fmt.Bshift = 8; fmt.Ashift = 0;
+    fmt.Rshift = 24;
+    fmt.Gshift = 16;
+    fmt.Bshift = 8;
+    fmt.Ashift = 0;
 #else
-  fmt.Rshift = 0; fmt.Gshift = 8; fmt.Bshift = 16; fmt.Ashift = 24;
+    fmt.Rshift = 0;
+    fmt.Gshift = 8;
+    fmt.Bshift = 16;
+    fmt.Ashift = 24;
 #endif
-  fmt.Rmask = 255 << fmt.Rshift;
-  fmt.Gmask = 255 << fmt.Gshift;
-  fmt.Bmask = 255 << fmt.Bshift;
-  fmt.Amask = 255 << fmt.Ashift;
-
-  SDL_Surface *tgt = DFSDL_ConvertSurface(src, &fmt, SDL_SWSURFACE);
-  DFSDL_FreeSurface(src);
-  for (int x = 0; x < tgt->w; ++x) {
-      for (int y = 0; y < tgt->h; ++y) {
-          Uint8* p = (Uint8*)tgt->pixels + y * tgt->pitch + x * 4;
-          if (p[3] == 0) {
-              for (int c = 0; c < 3; c++) {
-                  p[c] = 0;
-              }
-          }
-      }
-  }
-  return tgt;
+    fmt.Rmask = 255 << fmt.Rshift;
+    fmt.Gmask = 255 << fmt.Gshift;
+    fmt.Bmask = 255 << fmt.Bshift;
+    fmt.Amask = 255 << fmt.Ashift;
+
+    SDL_Surface* tgt = DFSDL_ConvertSurface(src, &fmt, SDL_SWSURFACE);
+    DFSDL_FreeSurface(src);
+    for (int x = 0; x < tgt->w; ++x) {
+        for (int y = 0; y < tgt->h; ++y) {
+            Uint8* p = (Uint8*)tgt->pixels + y * tgt->pitch + x * 4;
+            if (p[3] == 0) {
+                for (int c = 0; c < 3; c++) {
+                    p[c] = 0;
+                }
+            }
+        }
+    }
+    return tgt;
 }
 
-const uint32_t TILE_WIDTH_PX = 8;
-const uint32_t TILE_HEIGHT_PX = 12;
+// register surface in texture raws, get a texpos
+static long add_texture(SDL_Surface* surface) {
+    std::lock_guard<std::mutex> lg_add_texture(g_adding_mutex);
+    auto texpos = enabler->textures.raws.size();
+    enabler->textures.raws.push_back(surface);
+    return texpos;
+}
+
+TexposHandle Textures::loadTexture(SDL_Surface* surface) {
+    if (!surface)
+        return 0; // should be some error, i guess
 
-static size_t load_textures(color_ostream & out, const char * fname,
-                            long *texpos_start,
-                            int tile_w = TILE_WIDTH_PX,
-                            int tile_h = TILE_HEIGHT_PX) {
-    SDL_Surface *s = DFIMG_Load(fname);
-    if (!s) {
-        out.printerr("unable to load textures from '%s'\n", fname);
-        return 0;
+    auto handle = reinterpret_cast<TexposHandle>(surface); // not the best way? but cheap
```

Comment:
it might be better to just typedef TexposHandle as a SDL_Surface * in the header file and avoid the need to cast at all.